### ISSUE TITLE
Pre-approve users

### ISSUE THREAD

**Issue by rth** [paris-saclay-cds/ramp-board#424]
It would be nice to have a way to pre-approve users. 

For instance, for teaching events I typically have the list of 20-40 emails for all the participants. Then instead of clicking on "approve user" emails for each one of them, I could generate a list of email hashes and put it on the sever. When the participant signs up, if the email hash matches it would be auto-manically approved for the general sign-up and for the event (and if not the current procedure happens).

This could be either one extra DB table with columns ['email_hash', 'list of pre-approved events'] or even an optional file if we don't want to extend the BD schema.

**Comment by rth** [comment]
In its simplest version this could be just a new CLI command,
```
ramp-database pre-approve-users ./valid_user_hashes.csv
```
that would be run in a cron job every minute or so, query sign-up requests from the DB, and approve them when they are pre-approved.

**Comment by agramfort** [comment]
@kegl used to absorb the list of students from a csv and generate passwords
sent automatically by email

maybe this option is still available somewhere

>

**Comment by tomMoral** [comment]
Creating users and adding them to one event  based on an email list would be very helpful for teaching.
Even without sending password, they can always use the recovery procedure to get the password latter.

This is very annoying to have 30+ student sign up as there are always people signing up late, not following and it takes a lot of bandwidth to accept them in 2 steps (for RAMP and then the event).

cc @glemaitre with whom we discussed this feature today :grin:

**Comment by rth** [comment]
> Creating users and adding them to one event based on an email list would be very helpful for teaching.

One issue with this is that it would be not GDPR compliant (https://github.com/paris-saclay-cds/ramp-board/issues/440), as one need user consent to put private information (here email address) into the RAMP DB. Even for teaching. If users sign up, they give a form of week consent. Ideally there should be a checkbox at sign up to indicate that they accept term of use (mechanism done in https://github.com/paris-saclay-cds/ramp-board/pull/456).

That's why I think having some script that runs in cron, and queries DB for users that just signed up and checks/validates against a pre-defined list of emails (or their ideally hashes) would likely be better in that respect.

**Comment by kegl** [comment]
Yes, I used to have a command line command that signed up users from a csv file, this is easy to write. But I agree with @rth that we need an action from the users to accept terms and conditions. We could have an admin page where you can copy paste a list of mails. I wouldn't put them in the DB, rather in a temporary file (again because of GDPR), that would be stored for, say, a week.

**Comment by kegl** [comment]
Admin command would also work, or you wouldn't even need the command, just edit the temporary file on the server.
